lib/src/main/java/com/diffplug/spotless/Lint.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2022-2024 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Models a linted line or line range. Note that there is no concept of severity level - responsibility
+ * for severity and confidence are pushed down to the configuration of the lint tool. If a lint makes it
+ * to Spotless, then it is by definition.
+ */
+public final class Lint implements Serializable {
+	/** Any exception which implements this interface will have its lints extracted and reported cleanly to the user. */
+	public interface Has {
+		List<Lint> getLints();
+	}
+
+	/** An exception for shortcutting execution to report a lint to the user. */
+	public static class ShortcutException extends RuntimeException implements Has {
+		public ShortcutException(Lint... lints) {
+			this(Arrays.asList(lints));
+		}
+
+		private final List<Lint> lints;
+
+		public ShortcutException(Collection<Lint> lints) {
+			this.lints = List.copyOf(lints);
+		}
+
+		@Override
+		public List<Lint> getLints() {
+			return lints;
+		}
+	}
+
+	private static final long serialVersionUID = 1L;
+
+	private int lineStart, lineEnd; // 1-indexed, inclusive
```

Comment:
Why `1-indexed`?

Reply:
Because text editors are 1-indexed, VSCode doesn't have a line 0.

Reply:
Ok. I should have been slightly more verbose.

I was wrong *feeling* git diff was reporting 0-indexed line numbers. Trying to find some spec around it (`git diff` -> `hunk header` -> `linux patch`), the only reference to 1-indexed I found is:

> [...] apply at the start of the file if their first line number is 1

https://www.man7.org/linux/man-pages/man1/patch.1.html

Reply:
I'm confused. My sense is that line numbers are always 1-indexed when humans read them. Happy to discuss further if you would like a change.

Reply:
Sorry for the confusion, I'm totally fine with 1-indexed now it is crystal clear it is the standard. I just tried to find some references, as I initially thought 0-indexed would be more relevant

(I would close this conversation, but for any reason, the button is not appearing).